baram/ec2_manager.py

```diff
@@ -1,105 +1,274 @@
+            for sg_id in specified_sg_ids:
+                specified_enis = self.get_eni_with_sg_id(sg_id=sg_id)
+                pairs = {'vpc_id': vpc_id, 'sg_id': sg_id}
+                if len(specified_enis) != 0:
+                    for eni in specified_enis:
+                        pairs['eni_id'] = eni['NetworkInterfaceId']
+                        pairs['subnet_id'] = eni['SubnetId']
+                result.append(pairs)
+
+        return result
+
+    def get_sg_ids_with_vpc_id(self, vpc_id: str):
+        """
+        Get security group ids of specific vpc.
+
+        :param vpc_id: VpcId
+        :return: GroupId
+        """
+        sgs = self.cli.describe_security_groups()['SecurityGroups']
+        return [sg['GroupId'] for sg in sgs if sg['VpcId'] == vpc_id]
+
+    def get_eni_with_sg_id(self, sg_id: str):
+        """
+        Get network interfaces with specific security group.
+
+        :param sg_id: GroupId
+        :return: NetworkInterfaces
+        """
+        enis = self.cli.describe_network_interfaces()['NetworkInterfaces']
+        return [eni for eni in enis if eni['Groups'] != [] and sg_id in [x['GroupId'] for x in eni['Groups']]]
+
+    def list_sg_relations(self):
+        """
+        Describe security groups with related security groups, including egrss and ingress status.
+
+        :return: SecurityGroupId, IsEgress, RelatedSecurityGroupId
+        """
+        sg_rules = self.cli.describe_security_group_rules()['SecurityGroupRules']
+
+        sg_ids = [sg_rule['GroupId'] for sg_rule in sg_rules]
+        is_obs = [sg_rule['IsEgress'] for sg_rule in sg_rules]
+        referenced_sgs = [x['ReferencedGroupInfo']['GroupId']
+                          if 'ReferencedGroupInfo' in x.keys() else None for x in sg_rules]
+
+        result = []
+        for i in range(len(sg_rules)):
+            result.append({'sg_id': sg_ids[i],
+                           'is_ob': is_obs[i],
+                           'related_sg_id': referenced_sgs[i]})
+
+        return result
+
+    def get_related_sgs(self, sg_id: str):
+        """
+        Get the set of related security groups for specific security group.
+
+        :param sg_id: GroupId
+        :return: GroupId
+        """
+        sg_relations = self.list_sg_relations()
+        return set([sg_relation['sg_id'] for sg_relation in sg_relations if sg_relation['related_sg_id'] == sg_id])
+
+    def get_sg_rules(self, sg_id: str):
+        """
+        Get security group rules of specific security group.
+
+        :param sg_id: GroupId
+        :return: SecurityGroupRuleId
+        """
+        sg_rules = self.cli.describe_security_group_rules()['SecurityGroupRules']
+
+        result = [{'sg_rule_id': sg_rule['SecurityGroupRuleId'],
+                   'is_ob': True if sg_rule['IsEgress'] else False}
+                  for sg_rule in sg_rules if sg_rule['GroupId'] == sg_id]
+        return result
+
+    def revoke_sg_rules(self, sg_id: str):
```

Comment:
Isn't it better to use `delete` instead of `revoke`?

Reply:
I adopted aws-cli based representation `revoke_security_group_e(in)gress`. The word `delete` looks better for sustainability.

Reply:
It makes sense to follow convention but another method start with `delete`.

Is is also related to boto3 itsef?